node/entry/entry_gui.go

```diff
@@ -112,13 +112,30 @@ func Run() {
 		messagebox.Fatal("Exchange auth key failed", err.Error())
 	}
 
+	// Start the daemon HTTP server
+	go func() {
+		http.Handle("/local/auth/callback", http.HandlerFunc(app.onLoginCallback))
+		err := http.Serve(listener, nil)
+		if err != nil {
+			messagebox.Fatal("Serve listener failed", err.Error())
+		}
+	}()
+
+	zap.L().Info("Startup node successfully")
+
+	if err := app.Run(); err != nil {
+		messagebox.Fatal("Run application failed", err.Error())
+	}
+}
+
+func (app *osApp) Run() error {
 	dev, err := device.NewDevice()
```

Comment:
It's better to move L170 `app.run()` here to show tray first and then initialize the driver.

Reply:
Sure